src/Sentry/IMetricAggregator.cs

```diff
@@ -0,0 +1,107 @@
+    ///     The time when the metric was emitted. Defaults to the time at which the metric is emitted, if no value is provided.
+    /// </param>
+    /// <param name="stackLevel">Optional number of stacks levels to ignore when determining the code location</param>
+    void Increment(string key,
+        double value = 1.0,
+        MeasurementUnit? unit = null,
+        IDictionary<string, string>? tags = null,
+        DateTimeOffset? timestamp = null,
+        int stackLevel = 1);
+
+    /// <summary>
+    /// Emits a Gauge metric
+    /// </summary>
+    /// <param name="key">A unique key identifying the metric</param>
+    /// <param name="value">The value to be added</param>
+    /// <param name="unit">An optional <see cref="MeasurementUnit"/></param>
+    /// <param name="tags">Optional Tags to associate with the metric</param>
+    /// <param name="timestamp">
+    ///     The time when the metric was emitted. Defaults to the time at which the metric is emitted, if no value is provided.
+    /// </param>
+    /// <param name="stackLevel">Optional number of stacks levels to ignore when determining the code location</param>
+    void Gauge(string key,
+        double value = 1.0,
+        MeasurementUnit? unit = null,
+        IDictionary<string, string>? tags = null,
+        DateTimeOffset? timestamp = null,
+        int stackLevel = 1);
+
+    /// <summary>
+    /// Emits a Distribution metric
+    /// </summary>
+    /// <param name="key">A unique key identifying the metric</param>
+    /// <param name="value">The value to be added</param>
+    /// <param name="unit">An optional <see cref="MeasurementUnit"/></param>
+    /// <param name="tags">Optional Tags to associate with the metric</param>
+    /// <param name="timestamp">
+    ///     The time when the metric was emitted. Defaults to the time at which the metric is emitted, if no value is provided.
+    /// </param>
+    /// <param name="stackLevel">Optional number of stacks levels to ignore when determining the code location</param>
+    void Distribution(string key,
+        double value = 1.0,
+        MeasurementUnit? unit = null,
+        IDictionary<string, string>? tags = null,
+        DateTimeOffset? timestamp = null,
+        int stackLevel = 1);
+
+    /// <summary>
+    /// Emits a Set metric
+    /// </summary>
+    /// <param name="key">A unique key identifying the metric</param>
+    /// <param name="value">The value to be added</param>
+    /// <param name="unit">An optional <see cref="MeasurementUnit"/></param>
+    /// <param name="tags">Optional Tags to associate with the metric</param>
+    /// <param name="timestamp">
+    ///     The time when the metric was emitted. Defaults to the time at which the metric is emitted, if no value is provided.
+    /// </param>
+    /// <param name="stackLevel">Optional number of stacks levels to ignore when determining the code location</param>
+    void Set(string key,
+        double value = 1.0,
```

Comment:
Should this be an `int` instead? The `SetMetric` only deals with integers so this might be a bit misleading.

Reply:
Yeah, good call. I'll remove the default value as well (there's no reason for this to be arbitrarily 1 or 0 or anything else).